### ISSUE TITLE
If router configuration udpate fails, it is not reattempted

### ISSUE THREAD

**Issue by xaque208** [jeff-blank/otto#5]
Currently, the code is comparing only the public IP to the tunnel broker configuration.  This means that if the router configuration update fails, but the tunnel broker configuration succeeds, the router configuration remains incorrect and the tunnel is not established.

**Comment by jeff-blank** [comment]
I don't believe this is the case. if any part of the IP update fails, the state file is not updated; this causes the update steps to be reattempted at next timer tick. if you can provide specific examples of situations where this isn't the case, I'll take a look.

**Comment by xaque208** [comment]
Ah, it could be an issue with my state file.  I've only performed one update so far, so perhaps since my state file was incorrect, now that an update has been performed this won't be an issue any longer.  To create the initial state file I just created a file containing `{}`, which made the code run at least.  My first tailed update happened when my access class on the firewall was not correct.  I was hoping to narrow in a bit on the exact permissions that would be needed.

**Comment by jeff-blank** [comment]
until I address the initialization issues, I'd recommend seeding the state file with a public IP, like `{"pubIP":"127.0.0.1"}` or whatever value you like that's not going to show up on the public interface. I believe I had some issues as well using simply `{}` as an initial state file.
